### PR TITLE
Fix FUSE, fallback to tail, and add '-c' argument for shell

### PR DESCRIPTION
**Pull request by MikaelSmith**
Fix an issue with OSXFUSE where the mountpoint doesn't correctly unmount introduced by #303. Also make FUSE server shutdown more deterministic to avoid shutdown races.

Defer to `/usr/bin/tail` for plain `tail` calls to make it more useful.

Add `-c` argument for Wash shell, primarily for debugging.

### CODE REVIEW

File: cmd/rootMain.go [hunk 1]
```diff
@@ -117,6 +117,15 @@ func rootMain(cmd *cobra.Command, args []string) exitCode {
 		return exitCode{1}
 	}
 
+	// Mountpath is not cleaned up correctly if removed as part of deleting rundir, so it's placed
+	// in a separate location.
+	mountpath, err := ioutil.TempDir(cachedir, "mnt")
```
Comment (ekinanp) [inline cmd/rootMain.go]: Hmm, why'd you decide to leave https://github.com/puppetlabs/wash/blob/master/fuse/core.go#L156 alone (assuming that was the issue)? Just curious.

File: cmd/tail.go [hunk 2]
```diff
@@ -120,8 +121,19 @@ func tailMain(cmd *cobra.Command, args []string) exitCode {
 	}
 
 	if !follow {
-		cmdutil.ErrPrintf("Please use -f, other operations are not yet supported\n")
-		return exitCode{1}
+		// Defer to `/usr/bin/tail`
```
Comment (ekinanp) [inline cmd/tail.go]: Should this be documented somewhere?

File: cmd/rootMain.go (hunk 1 above)
Comment (MikaelSmith) [reply cmd/rootMain.go]: What do you mean "leave it alone"? I made some changes there, but it still didn't work. I couldn't find any leftover goroutines that might be finishing up there, so confused what's going on. May be an actual thread that the fuse library or libfuse is running that I can't find.

File: cmd/tail.go (hunk 2 above)
Comment (MikaelSmith) [reply cmd/tail.go]: Yeah, definitely. I'll add it to the long help.

File: cmd/rootMain.go (hunk 1 above)
Comment (ekinanp) [reply cmd/rootMain.go]: Sorry, that comment was made before I saw your next commit.

File: fuse/core.go [hunk 3]
```diff
@@ -164,17 +164,12 @@ func ServeFuseFS(filesys *plugin.Registry, mountpoint string) (chan<- context.Co
 		return nil, nil, err
 	}
 
-	// Start the FUSE server
+	// Start the FUSE server. We use the fuseExitedCh to catch externally triggered unmounts.
+	// If we're explicitly asked to shutdown the server, we want to wait until both Unmount and
+	// Serve have exited before signaling completion.
 	fuseServerStoppedCh := make(chan struct{})
```
[the remark this answers is not in the document]
Comment (MikaelSmith) [reply fuse/core.go]: Done.

File: cmd/rootMain.go [hunk 4]
```diff
@@ -14,6 +15,9 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// rootCommandFlag is associated with the `-c` option of the root command, set in root.go.
+var rootCommandFlag string
```
Comment (ekinanp) [inline cmd/rootMain.go]: Oh, was thinking in `root.go` b/c my first question was "Where is this defined?" when I saw it there. Guess a comment should go in both places, but I won't block the PR on it.

Comment (MikaelSmith) [reply cmd/rootMain.go]: Oh, ok.

**Pull request merged**